runtime/lib/backend/null_qubit/InstructionStrBuilder.hpp

```diff
@@ -0,0 +1,376 @@
+     * @brief Template specialized to return the string representation of complex numbers.
+     */
+    template <typename T> string element_to_str(const complex<T> &c)
+    {
+        ostringstream oss;
+        oss << c.real();
+
+        if (c.imag() != 0) {
+            // to keep printing output as short as possible, we only print the imaginary part
+            // whenever is different from zero;
+            oss << " + " << c.imag() << "i";
+        }
+        return oss.str();
+    }
+
+    /**
+     * @brief Takes as input a vector and returns its string representation. If with_brackets=True,
+     * the square brackets will be included to enclose the vector. E.g.
+     * - vector_to_string([0,1,2], false) returns "0, 1, 2"
+     * - vector_to_string([0,1,2], true) returns "[0, 1, 2]"
+     */
+    template <typename T>
+    string vector_to_string(const vector<T> &v, const bool &with_brackets = true)
+    {
+        ostringstream oss;
+        if (with_brackets)
+            oss << "[";
+
+        for (size_t i = 0; i < v.size(); i++) {
+            if (i > 0) {
+                oss << ", ";
+            }
+            oss << element_to_str(v[i]);
+        }
+        if (with_brackets)
+            oss << "]";
+
+        return oss.str();
+    }
+
+    /**
+     * @brief Builds the string representation of DataView
+     */
+    template <typename T, size_t R> string get_dataview_str(DataView<T, R> &dataview)
+    {
+        ostringstream oss;
+        bool is_first = true; // boolean to help determine where to put commas
+
+        oss << "[";
+        for (auto it = dataview.begin(); it != dataview.end(); it++) {
+            if (!is_first) {
+                oss << ", "; // if is not the first element then we add a comma to separate the
+                             // elements
+            }
+            oss << element_to_str(*it);
+            is_first = false;
+        }
+        oss << "]";
+        return oss.str();
+    }
+
+  public:
+    InstructionStrBuilder() = default;
+    ~InstructionStrBuilder() = default;
+
+    /**
+     * @brief This method is used to build a string representation of AllocateQubit(),
+     * AllocateQubits(), ReleaseQubit(), ReleaseAllQubits(), Measure()
+     */
+    template <typename T> string get_simple_op_str(const string &name, const T &param)
+    {
+        ostringstream oss;
+        oss << name << "(" << param << ")";
+        return oss.str();
+    }
+
+    /**
+     * @brief This method is used to build a string representation of Expval() and Var()
+     */
+    string get_simple_op_str(const string &name, const ObsIdType &o)
+    {
+        ostringstream oss;
+        oss << name << "(" << get_obs_str(o) << ")";
+        return oss.str();
+    }
+
+    /**
+     * @brief This method is used to build a string representation of State(), Probs(),
+     * PartialProbs()
+     */
+    template <typename T, size_t R>
+    string get_simple_op_str(const string &name, DataView<T, R> &dataview,
```

Comment:
These two functions have the same name as the first. Even though the signature is different, it would be nice if the name also reflected the intent :)

Reply:
18653b574f2dbdfc319f35d7b8a439a552d0a3d9